### ISSUE TITLE
Will it be possible to get something like this?

### ISSUE THREAD

**Issue by vineethvijayan** [i-schuetz/SwiftCharts#47]
![cbb5e934-70d1-11e5-829c-e5c857c34df4](https://cloud.githubusercontent.com/assets/6049347/10453497/2b80ca02-71ca-11e5-8fcd-e6a3863656d4.png)

I am trying to implement some thing like the above image, will it be possible?

**Comment by i-schuetz** [comment]
Hi,
yes it is, take a look at the multiaxis example. For the curved lines you can use layer of the cubic line example. For the little circles on the lines you can copy the layer from show touch coords example. You also find dividers and straight guidelines in some of the examples.

**Comment by vineethvijayan** [comment]
Yes, I looked at the multi axis, but I was not able to hide the y axis, it always shows the both axis.

**Comment by i-schuetz** [comment]
Hide how exactly? In your image the axes are visible...

2015-10-14 6:04 GMT+02:00 vineethvijayan <notifications@github.com>:

**Comment by vineethvijayan** [comment]
I got it working like this :+1: 

![img_0044](https://cloud.githubusercontent.com/assets/6049347/10479496/62d9a40c-7282-11e5-8bb7-f8ed1c36566c.PNG)

I also tried CubicLinePathGenerator 

![img_0042](https://cloud.githubusercontent.com/assets/6049347/10479501/6a4595fc-7282-11e5-8ed6-8d0600e9c73c.PNG)

but the data showing is wrong

**Comment by i-schuetz** [comment]
Hmm ok the curves need tuning, but besides of that what is wrong? Note also that you don't have generate the x axis values using the chart points, you can use regular intervals.

**Comment by vineethvijayan** [comment]
If I generate the X axis at regular intervals will it solve the cure issue?

**Comment by i-schuetz** [comment]
No the curve needs to be solved directly in the layer. Is this the only issue?

**Comment by vineethvijayan** [comment]
I am on the process of converting this as in the original pic. 

haven't faced anything else as of now.

**Comment by i-schuetz** [comment]
Ok I'll check the curves in the evening.

**Comment by vineethvijayan** [comment]
thanks mate :)

**Comment by i-schuetz** [comment]
Now it's possible to pass tensions points to the cubic line to control the curviness. 4e833839fff3918738b222ccdaf9d863902253bd 

You need 0.1 or so. You will have to use master to get this change. It's stable.

**Comment by vineethvijayan** [comment]
I am using 
pod 'SwiftCharts' 
in my pod file and I did 'pod update' still this change is not getting reflected

**Comment by vineethvijayan** [comment]
One more thing, is this something that is needed to be changed according to data? 
Asking this because the data will be dynamic

**Comment by i-schuetz** [comment]
`pod update` only works with released versions. You have to use 
```ruby
pod 'SwiftCharts', :git => 'https://github.com/i-schuetz/SwiftCharts.git'
```
in your podfile.
You don't have to change tension depending on data.

**Comment by vineethvijayan** [comment]
it looks okay but the curves are little bit sharp (irony)

![img_0045](https://cloud.githubusercontent.com/assets/6049347/10510567/7167ac10-7351-11e5-9b46-aee2b2739931.PNG)

**Comment by i-schuetz** [comment]
You can try out different tension values. I just committed something else which helps a bit with the sharp corners (doesn't affect the curviness of the lines though).

**Comment by vineethvijayan** [comment]
Sorry :-1:  

I tried ranging from 0 to 0.3 it only get worse :(

**Comment by i-schuetz** [comment]
Did you run `pod update` again to get the last commit? 

For further improvement as you see it's possible to pass your own curve generator. You can copy the current one and try to adjust it. I don't have time right now to continue with this.

Being practical - With 0.1 tension, the last commit, thicker lines and circles on the corners isn't it close enough to your initial image?

**Comment by vineethvijayan** [comment]
Yes I did, 

![img_0046](https://cloud.githubusercontent.com/assets/6049347/10511742/0d548344-7359-11e5-8c8c-2a6a14562803.PNG)

this is with tension 0.1

**Comment by i-schuetz** [comment]
Will try something, give me a few minutes.

**Comment by vineethvijayan** [comment]
Thanks mate and please take your time.

**Comment by i-schuetz** [comment]
Ok, I managed to get a curve like this 
![simulator screen shot 15 oct 2015 13 39 18](https://cloud.githubusercontent.com/assets/1381744/10512658/4ffdf17e-7342-11e5-88c1-e39bfb0da1ce.png)

with points `[(0, 1), (2, 16), (3, 8), (9, -16), (10, -56), (13, -80)]`, using `interpolateCGPointsWithCatmullRom`

The issue is that the current implementation has not real interpolation algorithm. I used this project https://github.com/jnfisher/ios-curve-interpolation Problem with this is it takes some time to add this to the library, for example, for some reason the first and last point are not shown in `interpolateCGPointsWithCatmullRom` so had to add filler points and `interpolateCGPointsWithHermite` looks weird.

For now I'd suggest that you pass your own path generator in which you use this project to interpolate the curve. Basically:

```swift
return UIBezierPath.interpolateCGPointsWithCatmullRom(points.map{NSValue(CGPoint: $0)}, closed: false, alpha: 1)
```
(or any other interpolation algorithm)

If you have more questions feel free to continue commenting this issue.


I just
